src/psyclone/psyir/backend/fortran.py

```diff
@@ -919,18 +924,21 @@ def gen_decls(self, symbol_table, is_module_scope=False):
             # already be done by the gen_use() method before)
             if isinstance(sym, ContainerSymbol):
                 all_symbols.remove(sym)
+                continue
             if sym.is_import:
                 all_symbols.remove(sym)
+                continue
             # All the IntrinsicSymbols (and unless we support all intrinsics,
```

Comment:
Please could you update this comment slightly - we need this irrespective of whether we support all intrinsics because Fortran will permit a Call to a routine that is only resolved at link time.

Reply:
Done